File: src/Java/Store/2. First Hole/src/main/refactoringgolf/store/Product.java
package refactoringgolf.store;

public class Product {

	/* The Name */
	public String name;

	/* The UnitPrice */
	public float unitPrice;

	/* The Category */
	public ProductCategory category;

	/* The Image */
	public ImageInfo image;

	/* The Category */
	public int unitsInStock;

	public Product(String name, float unitPrice, ProductCategory category, ImageInfo image) {
		this.name = name;
		this.unitPrice = unitPrice;
		this.category = category;
		this.image = image;
	}

	public String getName() {
		return name;
	}

	public float getUnitPrice() {
		return unitPrice;
	}

	public ProductCategory getCategory() {
		return category;
	}

	public ImageInfo getImage() {
		return image;
	}

	public int getUnitsInStock() {
		return unitsInStock;
	}

	public void setUnitsInStock(int unitsInStock) {
		this.unitsInStock = unitsInStock;
	}

	public String toXml() {
		return "<product>" + "<name>" + name + "</name>" + "<category>"
				+ category + "</category>" + "</product>";
	}
}

File: src/Net/1. First Course/1. Initial Tee/RefactoringGolf.Store/Product.cs
namespace RefactoringGolf.Store
{
    public class Product
    {
        /*The Name*/
        public string Name { get; private set; }

        /*The UnitPrice*/
        public decimal UnitPrice { get; private set; }

        /*The Category*/
        public ProductCategory Category { get; private set; }

        /*The Image*/
        public ImageInfo Image { get; private set; }

        /*The Category*/
        public int UnitsInStock { get; set; }

        public Product(string name, decimal unitPrice, ProductCategory category, ImageInfo image)
        {
            this.Name = name;
            this.UnitPrice = unitPrice;
            this.Category = category;
            this.Image = image;
        }

        public string ToXml()
        {
            return "<product>" +
                        "<name>" + Name + "</name>" +
                        "<category>" + Category + "</category>" +
                    "</product>";
        }
    }
}